src/core/reactor.cc

```diff
@@ -4173,39 +4174,45 @@ unsigned smp::adjust_max_networking_aio_io_control_blocks(unsigned network_iocbs
     auto aio_max_nr = read_first_line_as<unsigned>("/proc/sys/fs/aio-max-nr");
     auto aio_nr = read_first_line_as<unsigned>("/proc/sys/fs/aio-nr");
     auto available_aio = aio_max_nr - aio_nr;
-    available_aio -= std::min(available_aio, reserve_iocbs);
     auto requested_aio_network = network_iocbs * smp::count;
-    auto requested_aio_other = (storage_iocbs + preempt_iocbs) * smp::count;
+    auto requested_aio_other = reserve_iocbs + (storage_iocbs + preempt_iocbs) * smp::count;
     auto requested_aio = requested_aio_network + requested_aio_other;
 
     seastar_logger.debug("Intended AIO control block usage:");
     seastar_logger.debug("");
-    log_aiocbs(log_level::debug, storage_iocbs, preempt_iocbs, network_iocbs);
+    log_aiocbs(log_level::debug, storage_iocbs, preempt_iocbs, network_iocbs, reserve_iocbs);
     seastar_logger.debug("");
     seastar_logger.debug("Available AIO control blocks = aio-max-nr - aio-nr = {} - {} = {}", aio_max_nr, aio_nr, available_aio);
 
     if (available_aio < requested_aio) {
+        auto get_smp_count = [ linear_over_cpus = available_aio - std::min(reserve_iocbs, available_aio) ] (unsigned network) -> unsigned {
+            return linear_over_cpus / (storage_iocbs + preempt_iocbs + network);
+        };
+
         if (available_aio >= requested_aio_other + smp::count) { // at least one queue for each shard
+            unsigned smp_count_recommended = get_smp_count(network_iocbs);
```

Comment:
Maybe we can move this line before the conditional for both branches.

Reply:
@tomershafir Yes, I had considered it. The reason I didn't do it is that, on the exception-throwing branch, if `smp_count_max==0`, then `smp_count_recommended` is never defined / computed. Hoisting the definition and the `get_smp_count(network_iocbs)` call above the conditional slightly pessimizes the exception-throwing branch (we'd be computing something that might never be needed).

Reply:
Not changed in v2 (I think my argument was solid, and anyway, you'll probably rework or eliminate that code in your upcoming PR).